### ISSUE TITLE
BCC and CC for messages templates

### ISSUE THREAD

**Issue by jonathan-dejong** [eventespresso/event-espresso-core#120]
Hi!

I have a suggestion for adding BCC and CC fields for all messages templates. Currently these only support the to and from settings. 

A client needs the ability to get a BCC for each registration to another email :) 

I think it would be a beneficial improvement for just about anyone using EE.. Imagine someone wanting to get a BCC for whenever a registration fails to make sure they can follow up with the customer (for example). 

Thanks!

**Comment by nerrad** [comment]
I get the idea, it's not something that has been highly requested though.

In the meantime would it be sufficient to add a filter for the email_messenger headers so that BCC's can be added on the fly for just the outgoing messages related to EE?

**Comment by jonathan-dejong** [comment]
Yeah that would suffice in the meantime :) 

For my specific client this is related to an issue where they want better control of what emails have been sent etc. But I think I should create a separate issue for such a discussion.

**Comment by nerrad** [comment]
as you can see I just pushed the filter to master branch.  Hope that works for this case in the meantime :)

Based on your latter comment, I don't know what "better control" means exactly, but you may be interested in taking a look at [BETA-4.9-messages-queue](https://github.com/eventespresso/event-espresso-core/tree/BETA-4.9-messages-queue) branch.  We're adding a messages queue system that will keep a record of every outgoing email and queues for delivery rather than on demand generation and sending.  If all goes well it will be released with 4.9 for EE.

**Comment by jonathan-dejong** [comment]
Thank you!
So I can use that filter in the next update of EE then :) 

They're essentially asking for a way to determine what emails have been sent out and to who. Like a history/record. I've taken a look at the branch but apart from looking through all the changes to each file it's hard for me to tell what exactly it'll do.. but from what you're telling me it sounds like what they want?

**Comment by nerrad** [comment]
yes it sounds like the messages queue system will be what they want :)

In the meantime have you hard of the WP Mail Log plugin?  It hooks into wp_mail and records EVERY outgoing email from the WordPress system.  We use that a lot for internal testing of our messages system (for email messenger) and it works well.  It might be something your clients can use in the meantime.

**Comment by jonathan-dejong** [comment]
That's great news then :) 
Will it be able to sort the email log by event, type of message etc? 

No I had not.. looking at it I think they need something a bit more sophisticated as they're not comfortable with checking a txt file but a quick search found me some alternatives that might work better in the meantime! Thank you very much for the tip

**Comment by nerrad** [comment]
wp mail log plugin is NOT a text file  It adds a UI to the WordPress admin with a really nice interface that lists all the emails that have been sent.  There's a number of such plugins on the repo so its possible you were looking at the wrong one.. here's a direct link: https://wordpress.org/plugins/wp-mail-logging/

Regarding the new messages queue feature.  Yes users are able to filter the messages listed and sort/search etc.

**Comment by jonathan-dejong** [comment]
Ahh I took your suggestion literately: https://sv.wordpress.org/plugins/wp-mail-log/
;) 

I tested that plugin and it looks decent but I'm also testing some others I found because I'd like the ability to restrict which mails are being logged (like by from address). It says there's filters for developers but I can't find any documentation about it which is kinda bummer.


The new messages queue/record seems awesome then! Can't wait for it to come out :)
